packages/location-management/src/components/LocationUnitView/index.tsx

```diff
@@ -0,0 +1,221 @@
+reducerRegistry.register(locationHierarchyReducerName, locationHierarchyReducer);
+
+const { getFilterParams } = OpenSRPService;
+
+export interface AntTreeProps {
+  title: JSX.Element;
+  key: string;
+  children: AntTreeProps[];
+}
+
+/** Function to Load selected location unit for details
+ *
+ * @param {TableData} row data selected from the table
+ * @param {string} accessToken - access token
+ * @param {Function} setDetail funtion to set detail to state
+ */
+export function loadSingleLocation(
+  row: TableData,
+  accessToken: string,
+  setDetail?: React.Dispatch<React.SetStateAction<LocationDetailData | 'loading' | null>>
+): void {
+  if (setDetail) setDetail('loading');
+  const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_UNIT_GET);
+  // eslint-disable-next-line @typescript-eslint/no-floating-promises
+  serve
+    .read(row.id, { is_jurisdiction: true })
+    .then((res: LocationUnit) => {
+      if (setDetail) setDetail(res);
+    })
+    .catch(() => sendErrorNotification('An error occurred'));
+}
+
+/** Gets all the location unit at geographicLevel 0
+ *
+ * @param {string} accessToken - Access token to be used for requests
+ * @returns {Promise<Array<LocationUnit>>} returns array of location unit at geographicLevel 0
+ */
+export async function getBaseTreeNode(accessToken: string) {
+  const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_UNIT_FINDBYPROPERTIES);
+  return await serve
+    .list({
+      is_jurisdiction: true,
+      return_geometry: false,
+      properties_filter: getFilterParams({ status: 'Active', geographicLevel: 0 }),
+    })
+    .then((response: LocationUnit[]) => response);
+}
+
+/** Parse the hierarchy node into table data
+ *
+ * @param {Array<ParsedHierarchyNode>} hierarchy - hierarchy node to be parsed
+ * @returns {Array<TableData>} array of table data
+ */
+export function parseTableData(hierarchy: ParsedHierarchyNode[]) {
+  const data: TableData[] = [];
+
+  hierarchy.forEach((location, i: number) => {
+    data.push({
+      id: location.id,
+      key: i.toString(),
+      name: location.label,
+      geographicLevel: location.node.attributes.geographicLevel,
+    });
+  });
+  return data;
+}
+
+/** Gets the hierarchy of the location units
+ *
+ * @param {Array<LocationUnit>} location - array of location units to get hierarchy of
+ * @param {string} accessToken - Access token to be used for requests
+ * @returns {Promise<Array<RawOpenSRPHierarchy>>} array of RawOpenSRPHierarchy
+ */
+async function getHierarchy(location: LocationUnit[], accessToken: string) {
+  const hierarchy: RawOpenSRPHierarchy[] = [];
+
+  for await (const loc of location) {
+    const serve = new OpenSRPService(accessToken, API_BASE_URL, LOCATION_HIERARCHY);
+    const data = await serve.read(loc.id).then((response: RawOpenSRPHierarchy) => response);
+    hierarchy.push(data);
+  }
+
+  return hierarchy;
+}
+
+export const LocationUnitView: React.FC = () => {
```

Comment:
If it's not a hustle, I would recommend renaming `LocationUnitView` to `LocationUnitList`. I mistook this component for the `LocationUnitDetail` one

Reply:
yes it does but i say what ever it is it should be uniform in all packages and projects